cpp/src/strings/utilities.cu

```diff
@@ -45,15 +45,14 @@ std::unique_ptr<string_view, std::function<void(string_view*)>>
         return nullptr;
     auto length = std::strlen(str);
 
-    char* d_str{};
-    RMM_TRY(RMM_ALLOC( &d_str, length, stream ));
-    CUDA_TRY(cudaMemcpyAsync( d_str, str, length,
-                              cudaMemcpyHostToDevice, stream ));
+    auto* d_str = new rmm::device_buffer(length, stream);
+    CUDA_TRY(cudaMemcpyAsync(d_str->data(), str, length,
+                              cudaMemcpyHostToDevice, stream));
     CUDA_TRY(cudaStreamSynchronize(stream));
 
-    auto deleter = [](string_view* sv) { RMM_FREE(const_cast<char*>(sv->data()),0); };
+    auto deleter = [d_str](string_view* sv) { delete d_str; };
```

Comment:
Is it possible that d_str's data could be used on a stream other than `stream` before this deleter is called?

Reply:
Turns out this function is never used. 
However, the doxygen comment for the stream parameter already suggests the right use:
` * @param[in] stream Stream to execute any device code against.`
So IMO this should be fine as-is.